mcs/class/System.Web/System.Web/HttpApplication.cs

```diff
@@ -1643,7 +1657,39 @@ public virtual void Init ()
 				return true;
 			}
 		}
-		
+
+		public static void RegisterModule (Type moduleType) {
+			HttpRuntimeSection config = (HttpRuntimeSection)WebConfigurationManager.GetSection ("system.web/httpRuntime");
+
+			if (!config.AllowDynamicModuleRegistration)
+			{
+				throw new InvalidOperationException ("The Application has requested to register a dynamic Module, but dynamic module registration is disabled in web.config.");
+			}
+
+			dynamicModuleManeger.Add (moduleType);
+		}
+
+
+		private HttpModuleCollection CreateDynamicModules () {
+			HttpModuleCollection modules = new HttpModuleCollection ();
+
+			foreach (var module in dynamicModuleManeger.LockAndGetModules()) {
+				IHttpModule httpModule = this.CreateModuleInstance(module.Type);
+				httpModule.Init(this);
+				modules.AddModule(module.Name, httpModule);
+			}
+			return modules;
+		}
+
+		private IHttpModule CreateModuleInstance (Type type)
```

Comment:
Please remove the 'private' access modifier here and on line 1673, methods are private by default